### PR TITLE
Warn if ReactNoop.act is not awaited

### PR DESCRIPTION
**Pull request by acdlite**
Based on #14952

Updates the noop renderer tests to conform to the changes coming in #14853. It overlaps slightly with that PR but it doesn't implement the logic to warn when you nest async `act` calls; it only adds a warning if you call `act` without awaiting it.

I'm not sure if we should merge this PR now or if we should wait for #14853 to land first. It may make sense to merge this one and then rebase Sunil's forthcoming changes on top of that.

In other words, @threepointone, feel free to ignore this PR if you like — it only exists to potentially make your life easier.

### CODE REVIEW

File: packages/react-noop-renderer/src/createReactNoop.js [hunk 1]
```diff
@@ -832,38 +830,52 @@ function createReactNoop(reconciler: Function, useMutation: boolean) {
...
+          if (
+            result !== undefined &&
+            result !== null &&
+            typeof result.then === 'function'
+          ) {
+            result.then(() => {
+              actDepth--;
+              if (actDepth === 0) {
+                // TODO: Should keep flushing until queue is empty
+                ReactNoop.unstable_flushWithoutYielding();
```
Comment (threepointone) [inline packages/react-noop-renderer/src/createReactNoop.js]: this would simplify tests too (which I assume was the reason you did this). til, thanks!

**Comment by acdlite** [comment]
Closing in favor of #14853
